relay-server/src/actors/processor.rs

```diff
@@ -944,10 +923,16 @@ impl EnvelopeProcessor {
         //
         // Neither ID can be available in proxy mode on the /store/ endpoint. This is not supported,
         // since we cannot process an envelope without project ID, so drop it.
-        let project_id = project_state
+        let project_id = match project_state
             .project_id
             .or_else(|| envelope.meta().project_id())
-            .ok_or(ProcessingError::MissingProjectId)?;
+        {
+            Some(project_id) => project_id,
+            None => {
+                envelope_context.reject(Outcome::Invalid(DiscardReason::Internal));
```

Comment:
This would not be strictly necessary, since `DiscardReason::Internal` is the default when dropping the envelope context. In a discussion with @jjbayer, we decided to make this explicit to show clear intent that this wasn't missed.